src/cls/IPM/Lifecycle/Base.cls

```diff
@@ -584,7 +584,7 @@ Method %Reload(ByRef pParams) As %Status
         set tSkipScoped = 'tDeveloperMode
         set tVerbose = $get(pParams("Verbose"))
         set tRoot = $get(pParams("RootDirectory"))
-        if (tRoot = "") && tDeveloperMode {
+        if (tRoot = "") && (tDeveloperMode || ($get(pParams("PackageDeployNamespace")) '= "")) {
```

Comment:
Can you explain why this is needed? From my understanding, PrepareDeploy() runs after reload and simply maps everything into the newly created deploy namespace (doesn't actually load anything in).

Reply:
Ah ok that makes sense then regarding PrepareDeploy. Follow up: do we want to reload all the resources if deploying in a separate namespace? I thought everything is just mapped via PrepareDeploy? Also, when using PrepareDeploy, does the export with dependencies stuff work i.e. are dependencies also available in the new namespace or just the module?

Yes when in non-dev mode, we should just do it from the existing namespace rather than creating a scratch namespace since the only reason to install a non-dev mode package and then run publish is in a CI pipeline.